### ISSUE TITLE
Platformtoolset not supported

### ISSUE THREAD

**Issue by rvdn1974** [SonarSource/sonarlint-visualstudio#767]
### Description

When we install sonarlint and open our solution, we get platformtoolset not supported. It appears that the code of sonarlint can't determine the platformtoolset while MSbuild and Visualstudio can do it

### Repro steps

We define the platformtoolset on a central place, so we can control the toolset usage. Attached is a zipfile with a demo environment. Open the test\mydemo\ConsoleApplication1\ConsoleApplication1.sln. Select a cpp file and notice that sonarlint is complaining about unsupport platformtoolset
[test.zip](https://github.com/SonarSource/sonarlint-visualstudio/files/2600818/test.zip)
 ### Expected behavior

That sonarlint detects the platformtoolset correctly, or that we can set the default platformtoolset somewhere in sonarlint

### Actual behavior

System.ArgumentException: Unsupported platform

### Known workarounds

None

### Related information

* SonarLint for VS Version 4.7
* Visual Studio Version 2017 Enterprise update 8
* SonarQube server Version 6.7.5 (build 38563)
* Languages in your solution (c++)

**Comment by rvdn1974** [comment]
[test.zip](https://github.com/SonarSource/sonarlint-visualstudio/files/2600822/test.zip)

**Comment by duncanp-sonar** [comment]
@rvdn1974,

I can't repro the bug using the attached test project; .cpp file are analysed without any errors.

Could you add the error message from the SonarLint output window to the bug report please?

Currently SonarLint only recognizes a [fixed list](https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/SonarLintDaemon/CFamily.cs#L311) of platform toolsets. I can make the SonarLint analysis fail by changing _BuildProcessTemplates\Imports.props_ to specify an unrecognized value, but in that case the VS Solution output pane also has a message complaining that VS doesn't recognize the toolset (e.g. _"...\ConsoleApplication1.vcxproj : warning  : The build tools for v999 cannot be found. Install v999 to build using the v999 build tools."_)

Which platform toolset are you using?

**Comment by duncanp-sonar** [comment]
Update: I've created ticket #804 for the specific case where the _PlatformToolset_ property is not set, either in the main project file or in an _Imports.props_ file.

However, as I said above both SonarLint and the build work correctly for me with your test solution.
The only way I can get SonarLint to fail is to open the _Project Properties_ dialogue and delete the value in the _Platform Toolset_ field.

That causes VS to write an empty _PlatformToolset_ property value into the main project file i.e.
```
  <PropertyGroup Condition= ...>
    ...
    <PlatformToolset />
    ...
  </PropertyGroup>
```

This overwrites the value set in the _Imports.props_ file, which will cause SonarLint to fail with the "Unsupported platform: " message. However, looking at the diagnostic build output shows that MSBuild isn't using the toolset specified in _Imports.props_ either;  the toolset has been set to an empty value in the main project, so the standard Microsoft targets will fall back on a default toolset. On my machine, the build was using the toolset _v100_, not the _v141_ specified in the _Imports.props_ file.

You might want to check that your _vcxproj_ files don't contain an entry like `<PlatformToolset />` that would override the value set in your _Imports.props_ file.
